src/modules/job-list/job_util.c

```diff
@@ -91,9 +91,20 @@ static int store_attr (struct job *job,
         val = json_integer (job->ntasks);
     }
     else if (!strcmp (attr, "nnodes")) {
+        if (!(job->states_mask & FLUX_JOB_STATE_RUN)) {
+            /* job->nnodes potentially available in state DEPEND if
+             * resource specifically requested by user and is listed
+             * in jobspec.
+             */
+            if ((job->states_mask & FLUX_JOB_STATE_DEPEND)
+                && job->nnodes > 0) {
+                val = json_integer (job->nnodes);
+                goto out;
+            }
+            return 0;
+        }
```

Comment:
This should probably be obvious, but why is this extra block needed? Why can't you just always return `nnodes` if it is `> 0`, now that it may be `> 0` even before `RUN` state?

Reply:
So right now `nnodes` is re-calculated after it starts running, so `nnodes` can change if something goes wrong during the run.  Not sure if that is wise or not.

If `nnodes > 0` should `nnodes` not be re-calculated?  If that's probably for the best, then the above can just become `if nnodes > 0`, as you suggest.

Reply:
Ah, sorry, maybe this is because I don't know the code well enough.

I thought store_attr() was called from a job-list request to create the JSON response payload for each struct job *, and that if ``nodeschanges, thenjob->nnodes` would be updated, so you could always just grab `job->nnodes` and always get the "updated" value.

I think it is good that nnodes is getting updated at the RUN state.

Reply:
oh wait, you're right ... the check shouldn't matter, b/c the job state is irrelevant, `nnodes` is updated the regardless.  The check for job state is simply to see if the data is available / relevant.  doh!